org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/rename/RenameService2.xtend

```diff
@@ -197,11 +200,11 @@ class RenameService2 implements IRenameService2 {
 					if (element !== null && !element.eIsProxy) {
 						val leaf = NodeModelUtils.findLeafNodeAtOffset(rootNode, candidateOffset)
```

Comment:
Probably beyond this small bugfix here, but the rename infrastructure appears to be based on the assumption that IDs are comprised of leaf nodes and ignores the fact that `MyId: ID | 'someKeyword';` or `VersionedId: ID '-''(' INT ')';` are valid identifiers, too.

Reply:
The truth is, it doesn't even work with qualified names (as a datatype rule), unless you have registered a special value converter. My goal was to cover the basic behavior of the default ID rule. I think the method `doPrepareRename` is small enough to be overridden for more specific use cases.